tools/demo/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+project(demo)
+
+file(
+	GLOB_RECURSE
+	APP_SRC
+	src/*.h
```

Comment:
Although lambda and other tools follow this pattern, we shouldn't be using patterns like `src/*.h` & `GLOB_RECURSE` for adding files to our cmake builds as it's not a good practice and ultimately makes which files are being loaded obfuscated behind whatever patterns get matched. Could you change this to target the single `main.cpp` file here to start migrating towards concise file specifications?